contracts/WildcardSteward_v2.sol

```diff
@@ -122,22 +126,43 @@ contract WildcardSteward_v2 is Initializable {
         uint256[] memory tokens,
         address payable[] memory _benefactors,
         uint256[] memory _patronageNumerator,
-        uint256[] memory _tokenGenerationRate,
-        address[] memory _receiptERC20
+        uint256[] memory _tokenGenerationRate
     ) public onlyAdmin {
         assert(tokens.length == _benefactors.length);
+        assert(tokens.length == _patronageNumerator.length);
+        assert(tokens.length == _tokenGenerationRate.length);
+
         for (uint8 i = 0; i < tokens.length; ++i) {
-            // This index is starting from 0? How do we ensure non conflicting index to already minted tokens?
             assert(_benefactors[i] != address(0));
             benefactors[tokens[i]] = _benefactors[i];
             state[tokens[i]] = StewardState.Foreclosed; // TODO: Maybe Implement reverse dutch auction on intial sale
             patronageNumerator[tokens[i]] = _patronageNumerator[i];
-            tokenGenerationRate[tokens[i]] = _tokenGenerationRate[i]; // TODO: set these for old tokens
-            erc20Minter[tokens[i]] = IERC20Mintable(_receiptERC20[i]); // TODO: set for old tokens
-            emit AddToken(tokens[i], _patronageNumerator[i]);
+            tokenGenerationRate[tokens[i]] = _tokenGenerationRate[i];
+            emit AddToken(
+                tokens[i],
+                _patronageNumerator[i],
+                _tokenGenerationRate[i]
+            );
+        }
+    }
+
+    function addTokenGenerationRateToExistingTokens(
```

Comment:
There are no tests for this function. Not a big issue, we can test this via our migrations as a shortcut on testnet :)

Reply:
Yes, I guess testing it in the tests would involve us instating the tokens in the v1steward (where no rate exists), then upgrading to v2 and and adding a token generation rate to the token there? Not sure that step of switching from stewardv1 to v2 is possible in tests? We might want to try do something like this for our code coverage to remain 100%?

Reply:
Hmm, yeah, it won't be simple getting the token upgrade mechanism working in the tests I don't think. So we will test this manually in the migrations.